osu.Game.Rulesets.Osu/Mods/OsuModTransform.cs

```diff
@@ -59,9 +84,54 @@ private void applyTransform(DrawableHitObject drawable, ArmedState state)
                             .MoveTo(originalPosition, moveDuration, Easing.InOutSine);
                     }
 
-                    theta += (float)hitObject.TimeFadeIn / 1000;
+                    theta += (float)h.TimeFadeIn / 1000;
                     break;
             }
         }
+
+        private void applyEmitState(DrawableHitObject drawable)
+        {
+            switch (drawable)
+            {
+                case DrawableSliderHead:
+                case DrawableSliderTail:
+                case DrawableSliderTick:
+                case DrawableSliderRepeat:
+                    return;
+
+                default:
+                    var h = (OsuHitObject)drawable.HitObject;
+                    Vector2 origin = drawable.Position;
+
+                    using (drawable.BeginAbsoluteSequence(h.StartTime - h.TimePreempt))
+                    {
+                        drawable.ScaleTo(.6f).Then().ScaleTo(1, h.TimePreempt / 2, Easing.OutSine);
+                        drawable.MoveTo(OsuPlayfield.BASE_SIZE / 2).Then().MoveTo(origin, (h.TimePreempt / 2), Easing.Out);
+                    }
+
+                    break;
+            }
+        }
+
+        public override void ApplyToBeatmap(IBeatmap beatmap)
+        {
+            base.ApplyToBeatmap(beatmap);
+
+            foreach (var obj in beatmap.HitObjects.OfType<OsuHitObject>())
+                applyFadeInAdjustment(obj);
+
+            static void applyFadeInAdjustment(OsuHitObject osuObject)
+            {
+                osuObject.TimeFadeIn = osuObject.TimePreempt * .2;
+                foreach (var nested in osuObject.NestedHitObjects.OfType<OsuHitObject>())
+                    applyFadeInAdjustment(nested);
+            }
+        }
+    }
+
+    public enum MoveType
+    {
+        Rotate,
+        Emit
```

Comment:
Not sure about the naming here. "Emit" on its own doesn't say where the emission is coming from. Really, "Rotate" mode is also emitting from a point rotating around the object.

Reply:
How about "radiate"? i think that better illustrates the idea of notes being spewed in a circle. i cant think of anything else that's memorable and not annoyingly wordy like "centered note expulsion"